fuse/dl/templates/walkthrough_template.py

```diff
@@ -50,34 +48,37 @@
 # Example of a training template:
 #
 # Training template contains 6 fundamental building blocks:
-# (1) Data
+# (1) Data - detailed README file can be found at [fuse/data](../../fuse/data)
 # (2) Model
 # (3) Losses
-# (4) Metrics
+# (4) Metrics and Evaluation - detailed README file can be found at [fuse/eval](../../fuse/eval)
 # (5) Callbacks
 # (6) Manager
 #
-# The template will create each of those building and will eventually run Manager.train()
+# The template will create each of those building blocks and will eventually run Manager.train()
 #
 # Terminology:
-# 1.batch_dict -
-#   Dictionary that aggregates data and results during the batch cycle.
-#   The resulted batch is hierarchical dictionary including 3 branches: 'data', 'model', 'losses'
-#   batch_dict allows us to decouple the training components.
-#   For example 'classification_loss' code don't directly interact with the data or model,
-#   Instead a string name, which is a key in batch dict, will be specified for both the:
-#   (1) classification label: 'data.gt.global.label' (2) model logits predictions: 'model.head_0.logits'
-# 2.epoch_result -
-#   A dictionary created by the manager and includes the losses and metrics value
-#   as defined within the template and calculated for the specific epoch
-# 3.Fuse base classes - Fuse*Base -
-#   Abstract classes of the object forming together Fuse framework .
-# 4.Fuse default classes - Fuse*Default -
-#   A default generic implementation of the equivalent base class.
-#   Those generic implementation will be useful for most common use cases.
-#   Alternative implementations could be implemented for the special cases.
-# 5.sample_descriptor -
-#   Unique ID representing sample
+# 1. sample-id -
+#    A unique identifier of a sample. Each sample in the dataset must have an id that uniquely identifies it.
+#    Examples of sample ids:
+#    * path to the image file
+#    * Tuple of (provider_id, patient_id, image_id)
+#    * Running index
+# 2. sample_dict - 
+#    Represents a single sample and contains all relevant information about the sample.
+#    No specific structure of this dictionary is required, but a useful pattern is to split it into sections (keys that define a "namespace" ): such as "data", "model",  etc.
+#    NDict (fuse/utils/ndict.py) class is used instead of python standard dictionary in order to allow easy "." separated access. For example:
+#    `sample_dict[“data.input.img”]` is the equivalent of `sample_dict["data"]["input"]["img"]`
```

Comment:
is sample_dict also an NDict ? would "sample_dict.data.input.img" also work?

Reply:
Yes, sample_dict is NDict.
So it means that both sample_dict[“data.input.img”] and sample_dict["data"]["input"]["img"] are supported.
(Not sample_dict.data.input.img)